runtime/src/snapshot_utils.rs

```diff
@@ -206,11 +207,12 @@ fn get_archive_ext(archive_format: ArchiveFormat) -> &'static str {
 pub fn remove_tmp_snapshot_archives(snapshot_path: &Path) {
     if let Ok(entries) = fs::read_dir(&snapshot_path) {
         for entry in entries.filter_map(|entry| entry.ok()) {
-            if entry
+            let filename = entry
                 .file_name()
                 .into_string()
-                .unwrap_or_else(|_| String::new())
-                .starts_with(TMP_SNAPSHOT_PREFIX)
+                .unwrap_or_else(|_| String::new());
+            if filename.starts_with(TMP_SNAPSHOT_PREFIX)
+                | filename.ends_with(TMP_SNAPSHOT_EXTENSION)
```

Comment:
I'm a little uncomfortable with deleting *.tmp from wherever `--snapshot-dir` is aimed at, can you explain the issue that this line is aiming to resolve?

Reply:
tmp is extension of snapshot while it downloaded. If process was halted, than tmp file stays here forever. I saw 10 files one time)

When we use `--snapshot-dir` could are there other files with tmp extension in this separate folder?
But I don't sure that's true if we don't use `--snapshot-dir`.

You are right. Even if there are no these files now, but it could appear in future.

What about this?
```
const SNAPSHOT_PREFIX: &str = "snapshot-";

if filename.starts_with(TMP_SNAPSHOT_PREFIX)
    | filename.starts_with(SNAPSHOT_PREFIX) & filename.ends_with(TMP_SNAPSHOT_EXTENSION)
```

Reply:
What do you think about https://github.com/solana-labs/solana/pull/15939 as an alternative to changing `remove_tmp_snapshot_archives()`

Reply:
I like it. More universal.

Reply:
I have reverted this file to origin